ft-email-auth-provider/src/handlers/confirm_email.rs

```diff
@@ -0,0 +1,139 @@
+use crate::handlers::create_account::CreateAccount;
+use validator::ValidateEmail;
+
+#[ft_sdk::form]
+pub fn confirm_email(
+    mut conn: ft_sdk::Connection,
+    ft_sdk::Query(code): ft_sdk::Query<"code">,
+    ft_sdk::Query(email): ft_sdk::Query<"email">,
+) -> ft_sdk::form::Result {
+    if !email.validate_email() {
+        return Err(ft_sdk::single_error("email", "invalid email format").into());
+    }
+
+    let (id, data) = ft_sdk::auth::provider::user_data_by_custom_attribute(
+        &mut conn,
+        crate::PROVIDER_ID,
+        crate::EMAIL_CONF_CODE_KEY,
+        &code,
+    )?;
+
+    let sent_at = data
+        .custom
+        .as_object()
+        .expect("custom is a json object")
```

Comment:
Do not panic, return error.

Reply:
`custom` is only set in the code and read from the code, if it is not a json object then there must be something wrong in the implementation. So it makes sense to panic (use `expect`). 

Returning the error as it is will send the stack trace (wasm stack trace is not helpful anyway) which I don't like. Creating custom error does not make sense, what should be the error message?

Reply:
Now that I think about it, the host should never show the stack trace in browser (or show it in dev mode only) if the wasm functions returns an error. A generic 500 error would be better I think